janusgraph-hadoop/src/main/java/org/janusgraph/hadoop/compat/HadoopCompatLoader.java

```diff
@@ -0,0 +1,29 @@
+// Copyright 2017 JanusGraph Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.janusgraph.hadoop.compat;
+
+import org.janusgraph.hadoop.compat.h2.Hadoop2Compat;
+
+public class HadoopCompatLoader {
+
+
+    public static final HadoopCompat DEFAULT_COMPAT = getCompat();
+
+    // TODO add a string argument that allows specifying a class instead of relying heuristics around VersionInfo.getVersion()
+    // TODO add threadsafe caching that is aware of the string argument and instantiates a compat for each argument at most once (assuming the instantiation succeeds)
+    public static HadoopCompat getCompat() {
+        return new Hadoop2Compat();
```

Comment:
In the previous `HadoopCompatLoader` it was returning either `Hadoop1Compat` or `Hadoop2Compat` depending on `VersionInfo.getVersion()` and now it is always `Hadoop2Compat`. Is it OK?

Reply:
I will remove the compat completely. We only support hadoop2 for long time. I would prefer to support only tinkerpop supported version of Hadoop.

Reply:
I am OK dropping support for Hadoop 1 (if it isn't dropped yet) but is it a breaking change? If so, should this be reflected in `upgrade.adoc`?

Reply:
I will check the upgrade doc, but this pr will be followed by another PR to invert order of Hadoop and backends